src/Components/WebAssembly/WebAssembly.Authentication/src/Models/InteractiveRequestOptions.cs

```diff
@@ -0,0 +1,192 @@
+        AdditionalRequestParameters ??= new();
+        return AdditionalRequestParameters.TryAdd(name, value);
+    }
+
+    /// <summary>
+    /// Tries to remove an existing additional parameter.
+    /// </summary>
+    public bool TryRemoveAdditionalParameter<TValue>(string name)
+    {
+        ArgumentNullException.ThrowIfNull(name, nameof(name));
+        return AdditionalRequestParameters != null && AdditionalRequestParameters.Remove(name);
+    }
+
+    /// <summary>
+    /// Tries to retrieve an existing additional parameter.
+    /// </summary>
+    public bool TryGetAdditionalParameter<[DynamicallyAccessedMembers(JsonSerialized)] TValue>(string name, out TValue value)
+    {
+        ArgumentNullException.ThrowIfNull(name);
+
+        value = default;
+        if (AdditionalRequestParameters == null || !AdditionalRequestParameters.TryGetValue(name, out var rawValue))
+        {
+            return false;
+        }
+        if (rawValue is JsonElement json)
+        {
+            value = Deserialize(json);
+            AdditionalRequestParameters[name] = value;
+            return true;
+        }
+        else
+        {
+            value = (TValue)rawValue;
+            return true;
+        }
+
+        [UnconditionalSuppressMessageAttribute(
+            "Trimming",
+            "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+            Justification = "All the types serialized and deserialized have annotated APIs on the parent method that ensure the members are preserved.")]
+        static TValue Deserialize(JsonElement element) => element.Deserialize<TValue>();
+    }
+
+    [UnconditionalSuppressMessage(
+        "Trimming",
+        "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+        Justification = "Serializes 'InteractiveAuthenticationRequest' into a string")]
+    internal string ToState() => JsonSerializer.Serialize(this);
+
+    [UnconditionalSuppressMessage(
+        "Trimming",
+        "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+        Justification = "Deserializes InteractiveAuthenticationRequest")]
+    internal static InteractiveRequestOptions FromState(string state) => JsonSerializer.Deserialize<InteractiveRequestOptions>(state);
+
+    internal class Converter : JsonConverter<InteractiveRequestOptions>
+    {
+        [UnconditionalSuppressMessage(
+            "Trimming",
+            "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
+            Justification = "Serializes InteractiveAuthenticationRequest")]
```

Comment:
That's not really a justification :) But also, I don't see `RequiresUnreferencedCodeAttribute` on this method, so why is it necessary to suppress anything here? Same question on `Write` below.

Reply:
I have updated the attributes as long as the descriptions here. The culprit here is `JsonSerializer.Serialize` and I have just followed the same approach we do for other types we pass down to json.

Reply:
I switched to use the serialization context